tools/c7n_azure/c7n_azure/function_package.py

```diff
@@ -128,55 +129,56 @@ def get_function_config(self, policy, queue_name=None):
     def _get_policy(self, policy):
         return json.dumps({'policies': [policy]}, indent=2)
 
-    def _add_cffi_module(self):
-        """CFFI native bits aren't discovered automatically
-        so for now we grab them manually from supported platforms"""
+    def _update_perms_package(self):
+        os.chmod(self.pkg.path, 0o0644)
 
-        self.pkg.add_modules('cffi')
+    def build(self, policy, queue_name=None, extra_modules=[], extra_non_binary_packages=[]):
 
-        # Add native libraries that are missing
-        site_pkg = FunctionPackage._get_site_packages()[0]
+        c7n_azure_root = os.path.dirname(__file__)
+        cache_folder = os.path.join(c7n_azure_root, 'cache')
+        wheels_folder = os.path.join(cache_folder, 'wheels')
+        wheels_install_folder = os.path.join(cache_folder, 'dependencies')
 
-        # linux
-        platform = sys.platform
-        if platform == "linux" or platform == "linux2":
-            for so_file in os.listdir(site_pkg):
-                if fnmatch.fnmatch(so_file, '*ffi*.so*'):
-                    self.pkg.add_file(os.path.join(site_pkg, so_file))
+        modules = ['c7n', 'c7n-azure'] + extra_modules
+        non_binary_packages = \
+            ['pyyaml~=3.13', 'pycparser', 'tabulate>=0.8.2'] + extra_non_binary_packages
```

Comment:
It feels a little weird to have these pinned versions defined here in the middle of function_package.py?  Should we do something to make these arrays a little more manageable and maintainable or is there not much we can do?

Reply:
I believe you can use pkg_resources to inspect the dependencies of installed packages for exact versions. Ie pipdeptree or tl.eggdeps